libs/sdk-ui-kit/src/Dialog/EmbedInsightDialog/EmbedInsightDialogBase/components/HeightSetting.tsx

```diff
@@ -0,0 +1,87 @@
+// (C) 2022 GoodData Corporation
+import React, { useCallback } from "react";
+import { SingleSelectListItem } from "../../../../List";
+import { Dropdown, DropdownButton, DropdownList } from "../../../../Dropdown";
+import { NumericInput } from "./NumericInput";
+import { DEFAULT_UNIT, UNITS, UnitsType } from "../types";
+import { getDefaultHeightByUnit } from "../../utils";
+
+/**
+ * @internal
+ */
+export interface IHeightSettingProps {
+    value?: string;
+    unit?: UnitsType;
+    onValueChange: (value: string, unit: UnitsType) => void;
+}
+
+/**
+ * @internal
+ */
+export const HeightSetting: React.VFC<IHeightSettingProps> = (props) => {
+    const { value, onValueChange, unit = DEFAULT_UNIT } = props;
+
+    const onChange = useCallback(
+        (val: string) => {
+            if (val != value) {
+                onValueChange(val, unit);
+            }
+        },
+        [value, unit, onValueChange],
+    );
+
+    const onUnitChange = useCallback(
+        (unit: UnitsType) => {
+            onValueChange(getDefaultHeightByUnit(unit), unit);
+        },
+        [onValueChange],
+    );
+
+    return (
+        <div className="height-setting-component">
+            <NumericInput value={value ?? getDefaultHeightByUnit(unit)} onValueChanged={onChange} />
+            <UnitSelector selectedUnit={unit} onSelectUnit={onUnitChange} />
+        </div>
+    );
+};
+
+interface IDropdownItem {
+    id: UnitsType;
+    title: string;
+}
+
+interface UnitSelectorProps {
+    selectedUnit: UnitsType;
+    onSelectUnit: (unit: UnitsType) => void;
+}
+
+const UnitSelector: React.VFC<UnitSelectorProps> = (props) => {
+    const { selectedUnit, onSelectUnit } = props;
+    const items: IDropdownItem[] = UNITS.map((u) => ({ id: u, title: u }));
```

Comment:
Nitpick: you can call this outside of the render function, so it's not called on every re-render.

Reply:
Done